### ISSUE TITLE
Curiously kiwix-tools 3.3.0-1 was republished to https://download.kiwix.org/release/kiwix-tools/ 3 days ago on 2022-09-06

### ISSUE THREAD

**Issue by holta** [kiwix/kiwix-build#543]
Is this genuinely different from the 3.3.0-1 from 3 weeks ago, do you know?

![image](https://user-images.githubusercontent.com/2458907/189350358-abc46242-ee93-4985-85ca-fb4256ec3612.png)

(Background: https://github.com/iiab/iiab/pull/3344 !)

**Comment by kelson42** [comment]
@mgautierfr Any feedback?

**Comment by mgautierfr** [comment]
But the workflow itself is only triggered by a schedule or a tag `r_xx`

https://github.com/kiwix/kiwix-build/blob/754b716a23995bb57db5691666924a37d5f07239/.github/workflows/releaseNigthly.yml#L3-L8

You have workaround this by adding the `notarytool` branch as a trigger. And so the bug.

>  Given releasing is more sensitive, we should probably do the opposite: only release on specific conditions ; the most obvious ones would be a release event on github or a tag. What do you guys think?

Absolutely ! And we already have that https://github.com/kiwix/kiwix-build/blob/754b716a23995bb57db5691666924a37d5f07239/.github/scripts/common.py#L39
I will update the scripts.

---

@holta To answer your question, nothing has changed with this new upload.

**Comment by kelson42** [comment]
What is clear to me: we should never ever erase a binary which has already been published. This is a source of a lot of trouble.

**Comment by rgaudin** [comment]
> What is clear to me: we should never ever erase a binary which has already been published. This is a source of a lot of trouble.

Agreed. Maybe we should even enforce that at server level by *pushing* files to some staging area that then could run some checks before actually exposing them as releases. Could enable some security features like ensuring our binaries are signed.

**Comment by mgautierfr** [comment]
Full release process should probably introduce RC (Release candidate) and test on RC.
If RC is validated, archive should simply be renamed (not rebuild) and published.
